src/Orleans.Core/Configuration/Options/DnsNameGatewayListProviderOptions.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Collections.Generic;
+
+namespace Orleans.Configuration
+{
+    /// <summary>
+    /// Options for Configure DnsNameGatewayListProviderOptions
+    /// </summary>
+    public class DnsNameGatewayListProviderOptions
```

Comment:
Could we have a validator for this option, for example check if the port is not equal to `0` or check if the `DnsName` is resolvable?

Reply:
Validator for port < 0 completed.

I put in the DnsName resolvable validator but took it out.  The component already warns if DnsName is not resolvable and the silo handles the empty list accordingly.  The nice part is that it recovers very nicely once the DNS name is registered.

There are scenarios where the client component/frontend may complete deployment before the backend silo have not completed deployment and the DNS entries have not been registered.  If we throw in validation the client may crash on Init.

Reply:
The client should throw if there is no gateways available because the DNS isn't there. No problem with that. The client has an optional parameter to pass a delegate for retries so it can retry again. It is common to have the client initialize faster than silos.